core/src/tangram.cpp

```diff
@@ -500,7 +502,20 @@ const std::vector<TouchItem>& pickFeaturesAt(float _x, float _y) {
                                         _x, _y);
 }
 
+void queueSceneUpdate(const char* componentName, const char* value) {
+
+    return m_scene->queueComponentUpdate(componentName, value);
+}
+
+void applySceneUpdates() {
 
+    if (m_scene->userDefines().size() > 0) {
+        // reload the entire scene for now
+        loadScene(m_scene->path().c_str());
```

Comment:
So it looks like applying scene updates will load the scene file from the file system and parse the entire thing again, which can take multiple seconds and is not really necessary to this task. 

Instead we should probably be keeping a reference to the in-memory representation of the YAML document and simply applying the scene updates to that object.